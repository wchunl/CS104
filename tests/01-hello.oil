fail:     .function
          .param ptr expr
          .param ptr file
          .param int line
          .end
putchr:   .function
          .param int chr
          .end
putint:   .function
          .param int num
          .end
putstr:   .function
          .param ptr str
          .end
getchr:   .function int
          .end
getstr:   .function
          .end
getln:    .function
          .end
main:     .function int
          putstr("Hello, world!\n")
          return 0
          return 
          .end
