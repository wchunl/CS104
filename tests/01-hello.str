string_set[   0]:    4706206694200543916 0x1cc6c08->","
                   16579916199511856337 0x1cc8098->"getln"
string_set[   1]:    2297668033614959926 0x1cc88e8->"0"
string_set[   3]:   11868120863286193964 0x1cc82b8->"main"
                    9611428142580219387 0x1cc2608->"fail"
string_set[   4]:      21606623943142546 0x1cc8508->"{"
                    4017852018407011111 0x1cc6cf8->"file"
string_set[   6]:   13686721134187162920 0x1cc87b8->"}"
string_set[   8]:    9386170080689500517 0x1cc6ad8->"expr"
string_set[  12]:   15816506932648033945 0x1cc6968->"("
string_set[  17]:   18413295556186398326 0x1cc70f8->"putint"
string_set[  18]:    7211130878751977441 0x1cc7228->"putchr"
                   12679825168387489523 0x1cc7708->"num"
string_set[  23]:   11166701219533395393 0x1cc7e58->"getstr"
                   12868445110721718657 0x1cc6d18->")"
string_set[  28]:   14635384904416252381 0x1cc8848->"return"
string_set[  29]:   12291392672316406109 0x1cc6ea8->"line"
string_set[  31]:    6142509188972423790 0x1cc25e8->""
                   13348923631108115385 0x1cc74e8->"putstr"
string_set[  33]:    1583576014053361350 0x1cc7888->"str"
string_set[  35]:    8231216274321885541 0x1cc8748->""Hello, world!\n""
string_set[  38]:   12597418120160561823 0x1cc6b48->"int"
string_set[  39]:    7296690267450279874 0x1cc7368->"chr"
string_set[  40]:   11911796782241914524 0x1cc2218->"void"
string_set[  42]:    5344662657899890615 0x1cc6ec8->";"
string_set[  44]:   13497578857207247412 0x1cc6a08->"string"
string_set[  46]:    9201800968376702500 0x1cc7c38->"getchr"
load_factor = 0.574
bucket_count = 47
max_bucket_size = 2
