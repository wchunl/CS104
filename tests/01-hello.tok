#  1 "01-hello.oc"
#  1 "<built-in>"
#  1 "<command-line>"
#  1 "01-hello.oc"
#  1 "oclib.h"
# 22 "oclib.h"
   6 22.00001 261 TOK_VOID      void
   6 22.00006 282 TOK_IDENT     fail
   6 22.00011  40 '('           (
   6 22.00012 263 TOK_STRING    string
   6 22.00019 282 TOK_IDENT     expr
   6 22.00023  44 ','           ,
   6 22.00025 263 TOK_STRING    string
   6 22.00032 282 TOK_IDENT     file
   6 22.00036  44 ','           ,
   6 22.00038 262 TOK_INT       int
   6 22.00042 282 TOK_IDENT     line
   6 22.00046  41 ')'           )
   6 22.00047  59 ';'           ;
   6 24.00001 261 TOK_VOID      void
   6 24.00006 282 TOK_IDENT     putchr
   6 24.00013  40 '('           (
   6 24.00014 262 TOK_INT       int
   6 24.00018 282 TOK_IDENT     chr
   6 24.00021  41 ')'           )
   6 24.00022  59 ';'           ;
   6 25.00001 261 TOK_VOID      void
   6 25.00006 282 TOK_IDENT     putint
   6 25.00013  40 '('           (
   6 25.00014 262 TOK_INT       int
   6 25.00018 282 TOK_IDENT     num
   6 25.00021  41 ')'           )
   6 25.00022  59 ';'           ;
   6 26.00001 261 TOK_VOID      void
   6 26.00006 282 TOK_IDENT     putstr
   6 26.00013  40 '('           (
   6 26.00014 263 TOK_STRING    string
   6 26.00021 282 TOK_IDENT     str
   6 26.00024  41 ')'           )
   6 26.00025  59 ';'           ;
   6 28.00001 262 TOK_INT       int
   6 28.00005 282 TOK_IDENT     getchr
   6 28.00011  40 '('           (
   6 28.00012  41 ')'           )
   6 28.00013  59 ';'           ;
   6 29.00001 263 TOK_STRING    string
   6 29.00008 282 TOK_IDENT     getstr
   6 29.00014  40 '('           (
   6 29.00015  41 ')'           )
   6 29.00016  59 ';'           ;
   6 30.00001 263 TOK_STRING    string
   6 30.00008 282 TOK_IDENT     getln
   6 30.00013  40 '('           (
   6 30.00014  41 ')'           )
   6 30.00015  59 ';'           ;
#  5 "01-hello.oc"
   7 6.00001 262 TOK_INT       int
   7 6.00005 282 TOK_IDENT     main
   7 6.00009  40 '('           (
   7 6.00010  41 ')'           )
   7 6.00012 123 '{'           {
   7 7.00004 282 TOK_IDENT     putstr
   7 7.00011  40 '('           (
   7 7.00012 285 TOK_STRINGCON "Hello, world!\n"
   7 7.00029  41 ')'           )
   7 7.00030  59 ';'           ;
   7 8.00004 268 TOK_RETURN    return
   7 8.00011 283 TOK_INTCON    0
   7 8.00012  59 ';'           ;
   7 9.00001 125 '}'           }
