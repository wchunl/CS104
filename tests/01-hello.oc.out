; 0.  1: # 1 "01-hello.oc"; 1.  0: 
; 1.  1: # 1 "<built-in>"
; 2.  1: # 1 "<command-line>"
; 3.  1: # 1 "01-hello.oc"
; 4.  1: 
; 4.  2: 
; 4.  3: 
; 4.  4: # 1 "oclib.h" 1
; 5.  1: # 22 "oclib.h"
; 6. 22: void fail (string expr, string file, int line);
; 6. 23: 
; 6. 24: void putchr (int chr);
; 6. 25: void putint (int num);
; 6. 26: void putstr (string str);
; 6. 27: 
; 6. 28: int getchr();
; 6. 29: string getstr();
; 6. 30: string getln();
; 6. 31: # 5 "01-hello.oc" 2
; 7.  5: 
; 7.  6: int main() {
; 7.  7:    putstr ("Hello, world!\n");
; 7.  8:    return 0;
; 7.  9: }
hello